Corona-Warn-App/src/main/java/de/rki/coronawarnapp/presencetracing/risk/PresenceTracingRiskMapper.kt

```diff
@@ -0,0 +1,63 @@
+package de.rki.coronawarnapp.presencetracing.risk
+
+import de.rki.coronawarnapp.appconfig.AppConfigProvider
+import de.rki.coronawarnapp.appconfig.PresenceTracingRiskCalculationParamContainer
+import de.rki.coronawarnapp.risk.RiskState
+import de.rki.coronawarnapp.server.protocols.internal.v2.RiskCalculationParametersOuterClass
+import kotlinx.coroutines.flow.first
+import timber.log.Timber
+import javax.inject.Inject
+
+class PresenceTracingRiskMapper @Inject constructor(
+    private val configProvider: AppConfigProvider
+) {
+    private var presenceTracingRiskCalculationParamContainer: PresenceTracingRiskCalculationParamContainer? = null
+
+    suspend fun lookupTransmissionRiskValue(transmissionRiskLevel: Int): Double {
+        return getTransmissionRiskValueMapping()?.find {
+            (it.transmissionRiskLevel == transmissionRiskLevel)
+        }?.transmissionRiskValue ?: 0.0
+    }
+
+    suspend fun lookupRiskStatePerDay(normalizedTime: Double): RiskState {
+        return getNormalizedTimePerDayToRiskLevelMapping()?.find {
+            it.normalizedTimeRange.isInRange(normalizedTime)
+        }
+            ?.riskLevel
+            ?.mapToRiskState() ?: RiskState.CALCULATION_FAILED
+    }
+
+    suspend fun lookupRiskStatePerCheckIn(normalizedTime: Double): RiskState {
+        return getNormalizedTimePerCheckInToRiskLevelMapping()?.find {
+            it.normalizedTimeRange.isInRange(normalizedTime)
+        }
+            ?.riskLevel
+            ?.mapToRiskState() ?: RiskState.CALCULATION_FAILED
+    }
+
+    private suspend fun getTransmissionRiskValueMapping() =
+        getRiskCalculationParameters()?.transmissionRiskValueMapping
+
+    private suspend fun getNormalizedTimePerDayToRiskLevelMapping() =
+        getRiskCalculationParameters()?.normalizedTimePerDayToRiskLevelMapping
+
+    private suspend fun getNormalizedTimePerCheckInToRiskLevelMapping() =
+        getRiskCalculationParameters()?.normalizedTimePerCheckInToRiskLevelMapping
+
+    private suspend fun getRiskCalculationParameters(): PresenceTracingRiskCalculationParamContainer? {
```

Comment:
Should we maybe pass the config as parameter, so we have control over which config is used? When the calculation is called as part of some other code which might also use an app config for other purposes, so that at one instance in time, all the operations use the same config and we are not at the mercy of the caching mechanism always providing the same config. Could be the source of some difficult to reproduce bugs.

Reply:
You mean add a setter for the config? The config is only requested once per instance (unless there is none)